File: src/main/java/bank/InMemoryAccountRepository.java
package bank;

import java.util.HashMap;
import java.util.Optional;

public class InMemoryAccountRepository implements AccountRepository {
    public final HashMap<AccountId, Account> userBalance;

    public InMemoryAccountRepository() {
        userBalance = new HashMap<>();
    }

    @Override
    public Optional<Account> findById(AccountId accountId) {
        return Optional.ofNullable(userBalance.get(accountId));
    }

    @Override
    public void add(Account account) {
        userBalance.putIfAbsent(account.getAccountId(), account);
    }
}

File: Prevoir/InMemoryAccountRepository.cs
using System.Collections.Generic;

namespace Prevoir
{
    public class InMemoryAccountRepository: IAccountRepository
    {
        public InMemoryAccountRepository()
        {
            UserBalance = new Dictionary<AccountId, Account>();
        }

        public Dictionary<AccountId, Account> UserBalance { get; }

        public Account FindById(AccountId accountId)
        {
            UserBalance.TryGetValue(accountId, out var account);
            return account;
        }

        public void Add(Account account)
        {
            UserBalance.Add(account.Id, account);
        }

        public void Save(Account account)
        {
            UserBalance[account.Id] = account;
        }
    }
}